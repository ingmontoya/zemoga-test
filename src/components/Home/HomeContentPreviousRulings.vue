<script setup>
import { onMounted, ref } from 'vue';
import CustomSelect from '../CustomSelect.vue';
import useMobileDetection  from '@/composables/useMobileDetection';
import { useCustomSelectStore } from '@/stores/useCustomSelectStore';

const customSelectStore = useCustomSelectStore();
const { isMobile } = useMobileDetection();
const viewPicker = ref(['List', 'Grid']);
const selectedView = ref('');

function handleSelected(param) {
    selectedView.value = param;
}
</script>

<template>
    <aside class="content content-previous-rulings" role="doc-tip" aria-label="Previous Rulings">
        <h1 class="content__title"> Previous Rulings </h1>
        <CustomSelect v-if="!isMobile" :options="viewPicker" :default="selectedView || customSelectStore.selected" @input="handleSelected" class="content__viewPicker" />
    </aside>
</template>
