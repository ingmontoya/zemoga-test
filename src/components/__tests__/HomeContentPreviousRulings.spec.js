// HomeContentPreviousRulings.spec.ts
import { beforeEach, describe, it, expect } from 'vitest'
import { setActivePinia, createPinia } from 'pinia'
import { mount } from '@vue/test-utils'
import HomeContentPreviousRulings from '@/components/Home/HomeContentPreviousRulings.vue'

describe('HomeContentPreviousRulings.vue', () => {
  beforeEach(() => {
    const pinia = createPinia()
    setActivePinia(pinia)
  })

  it('renders correctly', () => {
    const wrapper = mount(HomeContentPreviousRulings, {
      global: {
        plugins: [createPinia()]
      }
    })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('updates selected view correctly', async () => {
    const wrapper = mount(HomeContentPreviousRulings, {
      global: {
        plugins: [createPinia()] // Asegúrate de que el store esté disponible para el componente
      }
    })

    await wrapper.find('.selected').trigger('click')
    await wrapper.findAll('.items div')[0].trigger('click')

    expect(wrapper.find('.selected').text()).toBe('List')
  })
})