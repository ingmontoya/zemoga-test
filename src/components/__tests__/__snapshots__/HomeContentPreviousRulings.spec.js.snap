// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HomeContentPreviousRulings.vue > renders correctly 1`] = `
<aside
  aria-label="Previous Rulings"
  class="content content-previous-rulings"
  role="doc-tip"
>
  <h1
    class="content__title"
  >
     Previous Rulings 
  </h1>
  <div
    class="content__viewPicker"
    tabindex="0"
  >
    <div
      class="selected"
    />
    <div
      class="items selectHide"
    >
      
      
    </div>
  </div>
</aside>
`;
